core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/OracleExportAction.java

```diff
@@ -0,0 +1,235 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.batch.action;
+
+import ch.ethz.ssh2.Connection;
+import ch.ethz.ssh2.Session;
+import ch.ethz.ssh2.StreamGobbler;
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Macro;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.action.Action;
+import co.cask.cdap.etl.api.action.ActionContext;
+import com.google.common.base.Charsets;
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.io.CharStreams;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Action that exports data from Oracle.
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name("OracleExport")
+@Description("Action to export data from Oracle")
+public class OracleExportAction extends Action {
+  private final OracleExportActionConfig config;
+  private StringBuilder scriptContent = null;
+  private String execute = null;
+  private static final Set<String> FORMATS = ImmutableSet.of("csv", "tsv", "psv");
+
+  public OracleExportAction(OracleExportActionConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void run(ActionContext context) throws Exception {
+    init();
+    Connection connection = new Connection(config.oracleServerHostname, config.oracleServerPort);
+    try {
+      connection.connect();
+      boolean isAuthenticated = connection.authenticateWithPassword(config.oracleServerUsername,
+                                                                    config.oracleServerPassword);
+      if (isAuthenticated == false) {
+        throw new IOException(String.format("SSH authentication error when connecting to %s@%s on port %d",
+                                            config.oracleServerUsername, config.oracleServerHostname,
+                                            config.oracleServerPort));
+      }
+      Session session = connection.openSession();
+      session.execCommand(execute);
+      InputStream stdout = new StreamGobbler(session.getStdout());
```

Comment:
input streams and buffered readers should be closed too. Please use try with resources syntax